drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1228 @@
+						 val + val_offset);
+		val_offset++;
+	}
+
+	if ((drv_cfg->axis & TMAG5273_MAG_CH_EN_Z) &&
+	    (all_mag_axis || (chan == SENSOR_CHAN_MAGN_Z))) {
+		tmag5273_channel_b_field_convert(drv_data->z_sample, drv_data->xyz_range,
+						 val + val_offset);
+		val_offset++;
+	}
+
+	if (drv_cfg->temperature && (chan == SENSOR_CHAN_DIE_TEMP)) {
+		tmag5273_temperature_convert(drv_data->temperature_sample, val + val_offset);
+		val_offset++;
+	}
+
+	if (drv_cfg->angle_magnitude_axis != TMAG5273_ANGLE_CALC_NONE) {
+		const bool all_calc_ch = (TMAG5273_CHAN_ANGLE_MAGNITUDE == (uint16_t)chan);
+
+		if (all_calc_ch || ((uint16_t)chan == SENSOR_CHAN_ROTATION)) {
+			tmag5273_angle_convert(drv_data->angle_sample, val + val_offset);
+			val_offset++;
+		}
+
+		if (all_calc_ch || ((uint16_t)chan == TMAG5273_CHAN_MAGNITUDE)) {
+			tmag5273_magnitude_convert(drv_data->magnitude_sample, drv_data->xyz_range,
+						   val + val_offset);
+			val_offset++;
+		}
+
+		if (all_calc_ch || (uint16_t)chan == TMAG5273_CHAN_MAGNITUDE_MSB) {
+			val[val_offset].val1 = drv_data->magnitude_sample;
+			val[val_offset].val2 = 0;
+
+			val_offset++;
+		}
+	}
+
+	if (val_offset == 0) {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief sets the \c DEVICE_CONFIG_1 and \c DEVICE_CONFIG_2 registers
+ *
+ * @param dev handle to the current device instance
+ *
+ * @retval 0 if everything was okay
+ * @retval -EIO on communication errors
+ */
+static inline int tmag5273_init_device_config(const struct device *dev)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+	int retval;
+	uint8_t regdata;
+
+	/* REG_DEVICE_CONFIG_1 */
+	regdata = 0;
+
+#if CONFIG_CRC
+	if (drv_cfg->crc_enabled) {
+		regdata |= TMAG5273_CRC_ENABLE;
+	}
+#endif
+
+	switch (drv_cfg->temperature_coefficient) {
+	case TMAG5273_DT_TEMP_COEFF_NDBFE:
+		regdata |= TMAG5273_MAGNET_TEMP_COEFF_NDBFE;
+		break;
+	case TMAG5273_DT_TEMP_COEFF_CERAMIC:
+		regdata |= TMAG5273_MAGNET_TEMP_COEFF_CERAMIC;
+		break;
+	case TMAG5273_DT_TEMP_COEFF_NONE:
+		__fallthrough;
+	default:
+		regdata |= TMAG5273_MAGNET_TEMP_COEFF_NONE;
+		break;
+	}
+
+	switch (drv_cfg->averaging) {
+	case TMAG5273_DT_AVERAGING_2X:
+		regdata |= TMAG5273_CONV_AVG_2;
+		break;
+	case TMAG5273_DT_AVERAGING_4X:
+		regdata |= TMAG5273_CONV_AVG_4;
+		break;
+	case TMAG5273_DT_AVERAGING_8X:
+		regdata |= TMAG5273_CONV_AVG_8;
+		break;
+	case TMAG5273_DT_AVERAGING_16X:
+		regdata |= TMAG5273_CONV_AVG_16;
+		break;
+	case TMAG5273_DT_AVERAGING_32X:
+		regdata |= TMAG5273_CONV_AVG_32;
+		break;
+	case TMAG5273_DT_AVERAGING_NONE:
+		__fallthrough;
+	default:
+		regdata |= TMAG5273_CONV_AVG_1;
+		break;
+	}
+
+	const int nb_captured_channels =
+		((drv_cfg->mag_channel >= TMAG5273_DT_AXIS_XYZ)
+			 ? 3
+			 : (((drv_cfg->mag_channel & TMAG5273_DT_AXIS_X) ? 1 : 0) +
+			    ((drv_cfg->mag_channel & TMAG5273_DT_AXIS_Y) ? 1 : 0) +
+			    ((drv_cfg->mag_channel & TMAG5273_DT_AXIS_Z) ? 1 : 0))) +
```

Comment:
nit, might be more readable this way? I'll let you be the judge tho

```suggestion
			 : POPCOUNT(drv_cfg->mag_channel)) +
```

Reply:
nah, nvm, didn't realize it would need to look at only the 3 least-significant bits, so not sure this would worth it in the or making things any easier to read...

Reply:
Well, using `POPCOUNT(drv_cfg->mag_channel & TMAG5273_DT_AXIS_XYZ)` would make it more readable, for sure. What do you think?

Reply:
Go for it :)